src/Grids/rectilinear_grid.jl

```diff
@@ -22,12 +22,14 @@ struct RectilinearGrid{FT, TX, TY, TZ, FX, FY, FZ, VX, VY, VZ, Arch} <: Abstract
       Δzᵃᵃᶜ :: FZ
       zᵃᵃᶠ  :: VZ
       zᵃᵃᶜ  :: VZ
-      # temporarly just to help refractoring with regular rectilinear grid (DELETE WHEN ALL THE Δx, Δy and Δz ARE REMOVED FROM THE CODE)
-      Δx :: FT
-      Δy :: FT
-      Δz :: FT
 end
 
+const XRegRectilinearGrid = RectilinearGrid{<:Any, <:Any, <:Any, <:Any, <:Number}
+const YRegRectilinearGrid = RectilinearGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Number}
+const ZRegRectilinearGrid = RectilinearGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Number}
+const HRegRectilinearGrid = RectilinearGrid{<:Any, <:Any, <:Any, <:Any, <:Number, <:Number}
+const  RegRectilinearGrid = RectilinearGrid{<:Any, <:Any, <:Any, <:Any, <:Number, <:Number, <:Number}
+
 function RectilinearGrid(FT = Float64;
```

Comment:
@simone-silvestri, I think we need a docstring here before merging this PR.

See, for example, the docstring for `RegularRectiliniarGrid`:
https://github.com/CliMA/Oceananigans.jl/blob/24e766481cebbc8f61099b386623d175218acedb/src/Grids/regular_rectilinear_grid.jl#L37-L153